File: vocalmind_2/vocalmind/types/vocal-dna.ts
// ─────────────────────────────────────────────
// 음색 DNA 타입 — 5축 보컬 특성 분석
// ─────────────────────────────────────────────

// ── Phase 13: 음색 DNA ──

export interface VocalDna {
  id: string;
  user_id: string;
  laryngeal: number;
  tongue_root: number;
  jaw: number;
  register_break: number;
  tone_stability: number;
  avg_pitch_hz: number | null;
  voice_type: string | null;
  source: string;
  created_at: string;
}

// VocalDNA alias (PascalCase 하위 호환)
export type VocalDNA = VocalDna;

export interface DnaAxis {
  key: keyof Pick<VocalDna, 'laryngeal' | 'tongue_root' | 'jaw' | 'register_break' | 'tone_stability'>;
  label: string;
  value: number;
}

File: vocalmind_2/vocalmind/backend/schemas/vocal_dna.py
"""보컬 DNA 분석 응답 스키마."""
from __future__ import annotations
from pydantic import BaseModel, Field


class VocalDnaResponse(BaseModel):
    laryngeal: float = Field(ge=0, le=100, description="후두 이완도 (높을수록 좋음)")
    tongue_root: float = Field(ge=0, le=100, description="혀뿌리 이완도")
    jaw: float = Field(ge=0, le=100, description="턱 이완도")
    register_break: float = Field(ge=0, le=100, description="성구전환 안정도")
    tone_stability: float = Field(ge=0, le=100, description="음색 안정도 (HNR 기반)")
    avg_pitch_hz: float | None = Field(default=None, description="평균 기본주파수 (Hz)")
    voice_type: str | None = Field(default=None, description="음역대 분류")
